### DISCUSSION TITLE
Is the "event not consumed" warning spurious?

### DISCUSSION THREAD

**Discussion by nicopap** [bevyengine/bevy#6595] [General]
```
Missed 31 `bevy_input::mouse::MouseMotion` events. Consider
reading from the `EventReader` more often (generally the best
solution) or calling Events::update() less frequently
(normally this is called once per frame). This problem is most
likely due to run criteria/fixed timesteps or consuming events
conditionally. See the Events documentation for
more information.
```

Live example: <https://nicopap.github.io/bevy_mod_paramap/earth3d.html> (open console, drag around the glob by holding middle click down)

Source code is: https://github.com/nicopap/bevy_mod_paramap/blob/7c5d9789ddd3cb05eb1b023cc0bab0aa5cfa021c/examples/earth3d.rs#L232-L308

I'm getting a warning when I'm doing something quite normal. I just do not read the `MouseMotion` event when I don't need to.

The error says `This problem is […] due to […] consuming events conditionally` but fails to explain why it is an issue. Looking at the [`Events` doc](https://docs.rs/bevy/latest/bevy/ecs/event/struct.Events.html), I don't see much of an argument against reading events conditionally.

It feels like a very common false positive. This is very confusing to a new user, especially given most of the warning message doesn't at all match what I'm actually doing in my code. In fact I do often skip reading events when I don't need to read them (for performance!) and I've seen this error in a lot of my projects.

I understand why it can help, but did we not forget to weight the cost of false positives when adding this feature? The likely thing I'll do next time I see this error is to just ignore it, and it might as well not exist in this case.

* PR introducing the message: https://github.com/bevyengine/bevy/pull/5730

**Comment by alice-i-cecile** [comment]
I'm running into this in #6589 as well. This warning is helpful for weird fixed timestep issues, but I agree: I'm not sure the cost is worth it here.
